tests/compile_upp.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+# Wen Meng 01/2020, Set up for cmake build.
+#############################################
+
+#set -eu
+
+#List of valid machines:
+validmachines=(wcoss_dell_p3 cray-intel hera orion jet)
+
+function usage {
+   echo "Usage:"
+   echo " $0 machinename"
+   echo ""
+   echo " Valid values for 'machinename' are: ${validmachines[@]}"
+   exit 1
+}
+
+if [ "$#" -eq 0 ]; then
+
+  #Check to see if we are building the old way
+  module purge
+
+  set -x
+  mac=$(hostname | cut -c1-1)
+  mac2=$(hostname | cut -c1-2)
+
+  if [ $mac2 = hf ] ; then   # For Hera
+    machine=hera
+  elif [ $mac = v -o $mac = m  ] ; then  # For WCOSS Dell
+    machine=wcoss_dell_p3
+  elif [ $mac = O ] ; then  # For orion
+    machine=orion
+  elif [ $mac = l -o $mac = s ] ; then  # For WCOSS Cray
```

Comment:
Detecting machine based on first and second character of the hostname is not a good idea.
If building on compute nodes, the hostnames are different.
An alternative is to check for machine specific paths e.g. `/scratch1` or `/gpfs` etc.
See UFS-weather-model for an example of to detect machines and load modules.